### DISCUSSION TITLE
Docker event stop channel being nil

### DISCUSSION THREAD

**Discussion by divyeshbhalekar** [slimtoolkit/slim#532] [Q&A]
I am trying to use github-action of docker-slim in my pipeline but facing this error 

dist_linux/
dist_linux/docker-slim
dist_linux/docker-slim-sensor
docker-slim: message='join the Gitter channel to ask questions or to share your feedback' info='https://gitter.im/docker-slim/community'
docker-slim: message='join the Discord server to ask questions or to share your feedback' info='https://discord.gg/9tDyxYS'
docker-slim: message='Github discussions' info='https://github.com/docker-slim/docker-slim/discussions'
cmd=build info=exec message='changing continue-after from probe to nothing because http-probe is disabled' 
cmd=build info=exec message='changing continue-after to enter' 
cmd=build state=started
cmd=build info=params target.type='image' target='gcr.io/***/aethia:93c0937' continue.mode='enter' rt.as.user='true' keep.perms='true' tags='gcr.io//:gcr.io/***/aethia:93c0937-slim' 
cmd=build state=image.inspection.start
cmd=build info=image id='sha256:dfe8d202e5b85a1989f6a1fbbb7531fee7ffdb8d02b88cec2e01a439dba2f979' size.bytes='60879508' size.human='61 MB' 
cmd=build info=image.stack index='0' name='gcr.io/***/aethia:93c0937' id='sha256:dfe8d202e5b85a1989f6a1fbbb7531fee7ffdb8d02b88cec2e01a439dba2f979' 
cmd=build state=image.inspection.done
cmd=build state=container.inspection.start
cmd=build info=container status='created' name='dockerslimk_4111_20230608091955' id='d46a6c0654c02294117659fe62be9efeebe4a3aaec156f3b3b0664de513b369d' 
cmd=build info=cmd.startmonitor status='sent' 
cmd=build info=event.startmonitor.done status='received' 
cmd=build info=container name='dockerslimk_4111_20230608091955' id='d46a6c0654c02294117659fe62be9efeebe4a3aaec156f3b3b0664de513b369d' target.port.list='' target.port.info='' message='YOU CAN USE THESE PORTS TO INTERACT WITH THE CONTAINER' 
cmd=build info=continue.after mode='enter' message='provide the expected input to allow the container inspector to continue its execution' 
cmd=build prompt='USER INPUT REQUIRED, PRESS <ENTER> WHEN YOU ARE DONE USING THE CONTAINER'
cmd=build state=container.inspection.finishing
cmd=build state=container.inspection.artifact.processing
cmd=build state=container.inspection.done
cmd=build state=building message=building optimized image 
cmd=build log='optimized.image.build' event=LOG.START tag='gcr.io//:gcr.io/***/aethia:93c0937-slim'  ====================
time="2023-06-08T09:19:59Z" level=fatal msg="docker-slim: failure" error="docker event stop chanel is nil" stack="goroutine 1 [running]:\nruntime/debug.Stack(0xc00051e1d0, 0x49f452, 0xc0000ac060)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/docker-slim/docker-slim/pkg/util/errutil.FailOn(0xfca5e0, 0xc0002b7ca0)\n\tgithub.com/docker-slim/docker-slim/pkg/util/errutil/errutil.go:28 +0x54\ngithub.com/docker-slim/docker-slim/pkg/app/master/inspectors/container.(*Inspector).FinishMonitoring(0xc00046d8c0)\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/inspectors/container/container_inspector.go:1089 +0x473\ngithub.com/docker-slim/docker-slim/pkg/app/master/commands/build.OnCommand.func2()\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/commands/build/handler.go:898 +0x85\ngithub.com/docker-slim/docker-slim/pkg/app.(*ExecutionContext).doCleanup(0xc00000d5e0)\n\tgithub.com/docker-slim/docker-slim/pkg/app/execontext.go:33 +0x5a\ngithub.com/docker-slim/docker-slim/pkg/app.(*ExecutionContext).Exit(0xc00000d5e0, 0x2000003)\n\tgithub.com/docker-slim/docker-slim/pkg/app/execontext.go:20 +0x2b\ngithub.com/docker-slim/docker-slim/pkg/app/master/commands/build.OnCommand(0xc00000d5e0, 0xc00007cc80, 0x7ffcadebc717, 0x1e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/commands/build/handler.go:1183 +0x7a45\ngithub.com/docker-slim/docker-slim/pkg/app/master/commands/build.glob..func1(0xc000568160, 0x0, 0xc00056c6f0)\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/commands/build/cli.go:629 +0x321b\ngithub.com/urfave/cli.HandleAction(0xd2e100, 0xeca9b0, 0xc000568160, 0xc000568160, 0x0)\n\tgithub.com/urfave/cli@v1.22.4/app.go:526 +0xfd\ngithub.com/urfave/cli.Command.Run(0xe824d9, 0x5, 0x0, 0x0, 0x1564aa0, 0x1, 0x1, 0xebe395, 0x6c, 0x0, ...)\n\tgithub.com/urfave/cli@v1.22.4/command.go:174 +0x58e\ngithub.com/urfave/cli.(*App).Run(0xc00007e1c0, 0xc0000ac120, 0x6, 0x6, 0x0, 0x0)\n\tgithub.com/urfave/cli@v1.22.4/app.go:279 +0x7e8\ngithub.com/docker-slim/docker-slim/pkg/app/master.Run()\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/app.go:15 +0x54\nmain.main()\n\tgithub.com/docker-slim/docker-slim/cmd/docker-slim/main.go:8 +0x25\n" version="linux|Transformer|1.37.3|97eb3c8c7f8c4e295a26c13882feb3f0539ab50f|2021-12-10_09:37:29AM"

cmd=build log='optimized.image.build' event=LOG.END tag='gcr.io//:gcr.io/***/aethia:93c0937-slim'  ====================
cmd=build info=build.error status='optimized.image.build.error' error='write unix @->/run/docker.sock: write: broken pipe' 
cmd=build state=exited code=33554435 version=linux|Transformer|1.37.3|97eb3c8c7f8c4e295a26c13882feb3f0539ab50f|2021-12-10_09:37:29AM location=/home/runner/work/aethia/aethia/dist_linux 
cmd=build state=container.target.shutdown.start
Error: Process completed with exit code 1.

my pipeline push stage 
Push_Artifact_to_Registry:
    if: (github.ref == 'refs/heads/master' && github.event_name == 'push') || (github.ref == 'refs/heads/main' && github.event_name == 'push') || (startsWith(github.ref, 'refs/tags/qa-') && github.event_name == 'push') || (startsWith(github.ref, 'refs/tags/prod-') && github.event_name == 'push')
    needs: Build_Artifact
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
        with:
          driver: docker
      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v4
        with:
          images: |
            ${{ env.REGISTRY_HOSTNAME }}/${{ env.IMAGE }}
          tags: |
            type=sha,enable=true,prefix=,suffix=,format=short
      - name: Login to GCR
        uses: docker/login-action@v2
        with:
          registry: gcr.io
          username: _json_key
          password: ${{ secrets.GCR_JSON_KEY }}
      - name: docker push
        uses: docker/build-push-action@v3
        id: push
        with:
          context: .
          push: true
          tags: ${{ steps.meta.outputs.tags }}
      - name: Minify
        run: |
          wget https://downloads.dockerslim.com/releases/1.37.3/dist_linux.tar.gz
          tar zxvf dist_linux.tar.gz
          chmod +x ./dist_linux/docker-slim
          SLIM_TAG="${{ steps.meta.outputs.tags }}-slim"
          ./dist_linux/docker-slim build --tag gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG} gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${{ steps.meta.outputs.tags }}
      - name: Inspect
        run: |
          docker image inspect gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG}
      - name: Push slim image to Google Container Registry
        run: |
          docker push gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG}

**Comment by kcq** [comment]
@divyeshbhalekar looks like you are using an old version (1.37.3). Try the current version (1.40.2): https://github.com/slimtoolkit/slim/releases/tag/1.40.2

The top level error in your post (`docker event stop chanel is nil`) is related to an old concurrency bug and shouldn't be there anymore. It is curious to see the broken pipe error with the docker socket, but let's see if you get more luck with the latest release first.

**Reply by divyeshbhalekar** [reply]
@kcq  still getting the error  i am using the version you specified (1.40.2):
My push stage :

  Push_Artifact_to_Registry:
    if: (github.ref == 'refs/heads/master' && github.event_name == 'push') || (github.ref == 'refs/heads/main' && github.event_name == 'push') || (startsWith(github.ref, 'refs/tags/qa-') && github.event_name == 'push') || (startsWith(github.ref, 'refs/tags/prod-') && github.event_name == 'push')
    needs: Build_Artifact
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
        with:
          driver: docker
      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v4
        with:
          images: |
            ${{ env.REGISTRY_HOSTNAME }}/${{ env.IMAGE }}
          tags: |
            type=sha,enable=true,prefix=,suffix=,format=short
      - name: Login to GCR
        uses: docker/login-action@v2
        with:
          registry: gcr.io
          username: _json_key
          password: ${{ secrets.GCR_JSON_KEY }}
      - name: docker push
        uses: docker/build-push-action@v3
        id: push
        with:
          context: .
          push: true
          tags: ${{ steps.meta.outputs.tags }}
      - name: Minify
        run: |
          wget https://downloads.dockerslim.com/releases/1.40.2/dist_linux.tar.gz
          tar zxvf dist_linux.tar.gz
          chmod +x ./dist_linux/docker-slim
          SLIM_TAG="${{ steps.meta.outputs.tags }}-slim"
          ./dist_linux/docker-slim build --tag gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG} --http-probe=false ${{ steps.meta.outputs.tags }}
      - name: Inspect
        run: |
          docker image inspect gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG}
      - name: Push slim image to Google Container Registry
        run: |
          docker push gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAM
[a.zip](https://github.com/slimtoolkit/slim/files/11703982/a.zip)
E }}:${SLIM_TAG}
          
          i have attached my entire pipeline

**Reply by kcq** [reply]
Now that we have the current version let's see what kind of errors you are getting now... It'll be good to add extra debug flags (global `--debug` flag and `--show-clogs` build command flag): `/dist_linux/slim --debug build --show-clogs --tag gcr.io/${{ env.GCP_PROJECT_ID }}/${{ env.SERVICE_NAME }}:${SLIM_TAG} --http-probe=false ${{ steps.meta.outputs.tags }}`

**Comment by divyeshbhalekar** [comment]
[8_Minify.txt](https://github.com/slimtoolkit/slim/files/11718887/8_Minify.txt)
i have added the log file & my pipeline
[a.zip](https://github.com/slimtoolkit/slim/files/11718905/a.zip)
